### PR TITLE
DEV-1124: holdings are equal if fields are equal or both empty

### PR DESCRIPTION
**Pull request by aelkiss**
This should address the underlying issue we were seeing with duplicate holdings not being cleaned up.

It probably would not hurt to add a test case that exercises the specific issue we were seeing -- create a cluster with a holding with some nil fields, update it with a holding with corresponding empty-string fields; ensure that they match and it doesn't end up with duplicate holdings. I will work on that next week, but wanted to get something up for a look now.

### CODE REVIEW

File: spec/clusterable/holding_spec.rb [hunk 1]
```diff
@@ -36,17 +36,42 @@
...
+      h2.n_chron = ""
+      h2.condition = ""
+      expect(h).to eq(h2)
     end
 
     (described_class.fields.keys - ["date_received", "uuid", "_id"]).each do |attr|
       it "== is false if #{attr} doesn't match" do
-        h2[attr] = nil
-        expect(h == h2).to be(false)
+        # ensure attribute in h2 is different from h but
```
Comment (aelkiss) [inline spec/clusterable/holding_spec.rb:61]: Before, we were always setting `h2[attr]` to nil, so the behavior change made this test fail -- now it sets it to an appropriate non-nil but non-equal type.

**Review by mwarin — APPROVED** [review]
Gah, I meant to actually approve not just say "APPROVE".
APPROVE.

**Comment by aelkiss** [comment]
@mwarin I added a test to cover the actual update situation, which revealed that for performance we were using `update_key` (a modified hash value) rather than equality checking in the holdings update. Please take a look and make sure this all still makes sense!

**Review by mwarin — APPROVED** [review]
Ran tests (green), looked at code (makes sense), saw new test (good).
APPROVE.
